src/pivot-table/hooks/use-selections-model.ts

```diff
@@ -16,25 +16,30 @@ export interface SelectedPivotCell {
   qCol: number;
 }
 
-export default function useSelectionsModel(selections: ExtendedSelections): SelectionModel {
+export default function useSelectionsModel(
+  selections: ExtendedSelections,
+  updatePageInfo: (args: Partial<PageInfo>) => void
+): SelectionModel {
   const isActive = selections.isActive();
   const [selected, setSelected] = useState<SelectedPivotCell[]>([]);
 
   useEffect(() => {
-    const clearSelections = () => setSelected([]);
-    const confirmOrCancelSelections = () => setSelected([]);
+    const clearSelections = () => {
+      setSelected([]);
+      updatePageInfo({ currentPage: 0 });
```

Comment:
I'm not sure this is the correct place to change the page. Some of these events do not update the data, then you shouldn't need to change page. Is it possible to look at if the data has actually changed and then reset the page?
We do that in sn-table

Reply:
There are 4 events that this `clearSelections` will run: 
1. deactivated
2. canceled
3. confirmed
4. cleared

I'll reset the page only in **confirmed** event then which we are sure that user has to see page 0.

Also in this level of component tree, we have access to the unprocessed data (data before separation + transformation into left, top and data grid), so Im not sure about checking/comparing data in that level.

Reply:
let me test it a bit, that might be good enough